File: Ballast.Client/src/messaging/events/game/game-state-changed.ts
import { EventBase } from '../../event-base';
import { Game } from 'ballast-core';

export class GameStateChangedEvent extends EventBase {

    public static readonly id: Symbol = Symbol.for('GameStateChangedEvent');

    public get id() {
        return GameStateChangedEvent.id;
    }

    public readonly game?: Game; 

    public constructor(game?: Game) {
        super();
        this.game = game;
    }

}

File: ballast-core/dotnet/src/application/events/GameStateChanged.cs
using Ballast.Core.Application.Models;
using Ballast.Core.Messaging;

namespace Ballast.Core.Application.Events
{
    public class GameStateChangedEvent : EventBase 
    {

        public static string GetId() => nameof(GameStateChangedEvent);
        public override string Id => GetId();

        public GameDto Game { get; private set; } 

        private GameStateChangedEvent() { } // For model binding / deserialization
        private GameStateChangedEvent(string eventDateIsoString, GameDto game) : base(eventDateIsoString)
        {
            Game = game;
        }

        public static GameStateChangedEvent FromGame(GameDto game) =>
            new GameStateChangedEvent(EventBase.GetDateIsoString(), game);

    }
}